modules/cli/src/main/java/org/apache/ignite/internal/cli/core/repl/completer/DynamicCompleterFactory.java

```diff
@@ -89,4 +93,13 @@ public LazyDynamicCompleter clusterConfigCompleter(Set<String> activationPrefixe
             }
         });
     }
+
+    public LazyDynamicCompleter nodeNameCompleter(String... activationPrefixes) {
+        return nodeNameCompleter(Set.of(activationPrefixes));
+    }
+
+    public LazyDynamicCompleter nodeNameCompleter(Set<String> activationPrefixes) {
+        return new LazyDynamicCompleter(() -> new NodeNameDynamicCompleter(activationPrefixes, nodeNameRegistry.getAllNames()));
```

Comment:
You should not use LazyDynamicCompleter for node name completions. There is no reason for that. Also, LazyDynamicCompleter will initialize only once and **always** provide the same set of node names.

Reply:
done